lago/export.py

```diff
@@ -0,0 +1,239 @@
+        """
+        with LogTask('Making disk sparse'):
+            # Removed unused space from the disk
+            utils.sparse(self.dst, self.disk['format'])
+
+    def calc_sha(self, checksum):
+        """
+        Calculate the checksum of the new exported disk, write it to
+        a file, and update this managers 'exported_metadata'.
+
+        Args:
+            checksum(str): The type of the checksum
+        """
+        with LogTask('Calculating {}'.format(checksum)):
+            with open(self.dst + '.hash', 'wt') as f:
+                sha = utils.get_hash(self.dst, checksum)
+                f.write(sha)
+            self.exported_metadata[checksum] = sha
+
+    def compress(self):
+        """
+        Compress the new exported image,
+        Block size was taken from virt-builder page
+        """
+        if not self.do_compress:
+            return
+        with LogTask('Compressing disk'):
+            utils.compress(self.dst, 16777216)
+            os.unlink(self.dst)
+
+    def update_lago_metadata(self):
+        with LogTask('Updating Lago metadata'):
+            self.exported_metadata['size'] = os.stat(self.dst).st_size
+            self.exported_metadata['name'] = self.name
+            self.exported_metadata['version'] = time.strftime("%Y%m%d.0")
+
+    def write_lago_metadata(self):
+        with LogTask('Writing Lago metadata'):
+            with open(self.dst + '.metadata', 'wt') as f:
+                json.dump(self.exported_metadata, f)
+
+
+class TemplateExportManager(DiskExportManager):
+    """
+    TemplateExportManager is responsible exporting images of type template.
+
+    Attributes:
+        standalone (bool): If true, create a new image which is the result of
+         merging all the layers of src (the image that we want to export).
+        src_qemu_info (dict): Metadata on src which was generated by qemu-img.
+    """
+
+    def __init__(self, dst, disk_type, disk, do_compress, *args, **kwargs):
+        super(TemplateExportManager, self).__init__(
+            dst, disk_type, disk, do_compress
+        )
+        self.standalone = kwargs['standalone']
+        self.src_qemu_info = utils.get_qemu_info(self.src, backing_chain=True)
+
+    def rebase(self):
+        """
+        Change the backing-file entry of the exported disk.
+        Please refer to 'qemu-img rebase' manual for more info.
+        """
+        if len(self.src_qemu_info) == 1:
+            # Base image (doesn't have predecessors)
+            return
+
+        if self.standalone:
+            # Consolidate the layers and base image
+            utils.qemu_rebase(target=self.dst, backing_file="")
+        else:
+            if len(self.src_qemu_info) > 2:
+                LOGGER.error(
+                    'Layered export is currently supported for one '
+                    'layer only.  You can try to use Standalone export.'
+                )
+                sys.exit(1)
```

Comment:
why exit here? better throw an exception and let it propagate(like in the rest of the code).